### PR TITLE
Remove deprecated types

### LINKED ISSUE #23051: `metabase-types/types` imports are deprecated
As `metabase-types/types` has been deprecated, we should not be using it anymore, neither in imports nor using the alias.

__TODOs__
- [ ] Replace all `metabase-types/types` imports with `metabase-types/api`
- [ ] Remove alias for `metabase-types/types`

### PR DESCRIPTION
**Pull request by kulyk**
Resolves #23051

Updates the codebase so deprecated `metabase-types/types` are no longer used and wipes the folder 🎉 
The final diff ended up being huge, it'd make sense to review things commit-by-commit (each commit does everything needed to remove exactly one file from `metabase-types/types`)

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30585)

### CODE REVIEW

**Comment by uladzimirdev** [comment]
you not only did it, but also made separated commits to simplify review 🏅 

I hope you'll get approvals soon, otherwise there should be conflicts with changes in master 😅

File: frontend/src/metabase/visualizations/components/ObjectDetail/utils.unit.spec.ts
Comment (alxnddr) [inline frontend/src/metabase/visualizations/components/ObjectDetail/utils.unit.spec.ts:56]: Seems like it is not a column but a `Field` since you assign `hasRemappedValue` below, wdyt about using the Field class here?

**Review by alxnddr — APPROVED** [review]
Looks great 🏅 
Left one question about columns vs fields, please have a look before merging

File: frontend/src/metabase-lib/mocks.ts
Comment (kulyk) [reply frontend/src/metabase-lib/mocks.ts:57]: Definitely not 😬 Nice catch, thank you!

**Pull request merged**
